### ISSUE TITLE
Running Electron App Doesn't Close After Uninstalling

### ISSUE THREAD

**Issue by darshitshah8** [electron-userland/electron-builder#8441]
I have developed an Electron.js application and successfully launched it. While the application is running in the background, I proceeded to uninstall it from my system. However, I've noticed that even after completing the uninstallation process, the application continues to run in the background. It doesn't automatically close or terminate, despite being uninstalled.

Is there a way to ensure that the running instance of the application closes automatically when the application is uninstalled? Any guidance or best practices on how to handle this scenario in Electron.js would be greatly appreciated.

**Comment by mmaietta** [comment]
No that's not possible AFAIK as that sounds like a OS/system limitation. The running app is in-memory and the files of the app are still being used by the application.

**Comment by darshitshah8** [comment]
Yes, the files are being used by the running application. I need a solution: how can I close the running application when the application is running and I uninstall the application?

**Comment by oi-pst** [comment]
I think it is related to [this](https://github.com/electron-userland/electron-builder/issues/7493).
Using electron-builder@22 will resolve the issue in most cases.
It's probably a bug.

**Comment by mmaietta** [comment]
@beyondkmp would you mind taking a look at this? I have zero familiarity with nsis scripts for windows, curious if you're able to get it set up and testable on your windows dev machine

**Comment by darshitshah8** [comment]
the issue is solved for me now , i have electron-builder@**24.13.3** and it working perfect for me, if anyone want to look at it. [find here](https://medium.com/@darshitshah8/how-to-close-a-running-electron-app-and-uninstall-appdata-using-nsis-on-uninstallation-dfb658b3853d)
